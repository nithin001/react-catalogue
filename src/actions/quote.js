import { put } from '../api/cart';
export const load = (cart) => {
  return (dispatch) => {
    dispatch({ type: 'LOAD_QUOTE_START' });
    return put(cart)
      .then((quote) => {
        dispatch({ type: 'POPULATE_QUOTE', payload: quote });
        dispatch({ type: 'LOAD_QUOTE_COMPLETE' });
      })
      .catch(() => {
        dispatch({ type: 'LOAD_ERROR' });
      });
  };
};