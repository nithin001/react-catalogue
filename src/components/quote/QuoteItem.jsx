import React from 'react';

export const QuoteItem = ({ order }) => (
  <div className={'quote-item'}>
    <span className={'quote-item__name'}>
      <span className={'quote-item__name quote-item__name--legend'}>Name:&nbsp;</span>
      <span className={'quote-item__name quote-item__name--value'}>{order.get('name')}</span>
    </span>
    <br/>
    <span className={'quote-item__price'}>
      <span className={'quote-item__name quote-item__name--legend'}>Price:&nbsp;</span>
      <span className={'quote-item__price quote-item__price--amount'}>{order.getIn(['price', 'amount'])}</span>
      &nbsp;
      <span className={'quote-item__price quote-item__price--currency'}>{order.getIn(['price', 'currency'])}</span>
    </span>
    <br/>
    <span className={'quote-item__total-quantity'}>
      <span className={'quote-item__quantity'}>
        <span className={'quote-item__quantity quote-item__quantity--legend'}>Quantity:&nbsp;</span>
        <span className={'quote-item__quantity quote-item__quantity--value'}>{order.get('quantity')}</span>
      </span>
      <span className={'quote-item__total'}>
        <span className={'quote-item__total quote-item__total--amount'}>{order.getIn(['lineTotal', 'amount'])}</span>
        &nbsp;
        <span className={'quote-item__total quote-item__total--currency'}>{order.getIn(
          ['lineTotal', 'currency'])}</span>
      </span>
    </span>
  </div>
);