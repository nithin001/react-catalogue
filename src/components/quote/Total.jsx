import React from 'react';

export const Total = ({ total }) => (
  <div className={'quote-total'}>
    <span className={'amount'}>{total.get('amount')}</span>
    <span className={'currency'}>{total.get('currency')}</span>
  </div>
);