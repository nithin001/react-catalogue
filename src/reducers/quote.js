import Immutable from 'immutable';
const quote = (state = Immutable.Map(), action) => {
  if (action.type === 'POPULATE_QUOTE') {
    return Immutable.fromJS(action.payload);
  }
  return state;
};
export default quote;