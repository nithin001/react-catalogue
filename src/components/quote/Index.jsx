import React from 'react';
import { connect } from 'react-redux';
import { Total } from './Total';
import { QuoteItem } from './QuoteItem';
export const Quote = ({ quote }) => {
  if (quote.get('lines') === undefined || quote.get('lines').size == 0) {
    return <span>Quotation is empty</span>;
  }
  const quoteItems = quote.get('lines').map(quoteItem => {
    return (<QuoteItem key={quoteItem.get('sku')} order={quoteItem}/>);
  });

  return (
    <div className={'quote'}>
      {quoteItems}
      <Total total={quote.get('total')}/>
    </div>
  );
};

export const mapStateToProps = (state) => {
  return { quote: state.quote };
};

export default connect(mapStateToProps, null)(Quote);
