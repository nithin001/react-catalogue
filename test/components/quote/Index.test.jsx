import React from 'react';
import renderer from 'react-test-renderer';
import quote from '../../../src/reducers/quote';
import { Quote } from '../../../src/components/quote';
var mockQuote = {
  'lines': [{
    'sku': '374847',
    'name': 'Nice Product',
    'price': {
      'amount': 20,
      'currency': 'EUR'
    },
    'quantity': 3,
    'lineTotal': {
      'amount': 60,
      'currency': 'EUR'
    }
  }],
  'total': {
    'amount': 60,
    'currency': 'EUR'
  }
};
const quoteState = quote(undefined, { type: 'POPULATE_QUOTE', payload: mockQuote });
describe('QuoteItem component', () => {
  it('should render the quote item with name, price, quantity and line total', () => {
    const props = {
      quote: quoteState
    };
    const quote = renderer.create(<Quote
      {...props}
    />);
    expect(quote).toMatchSnapshot();
  });
});