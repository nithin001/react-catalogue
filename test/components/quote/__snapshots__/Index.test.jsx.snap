// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component should render the quote items 1`] = `
<div
  className="quote"
>
  <div
    className="quote-item"
  >
    <span
      className="sku"
    >
      374847
    </span>
    <span
      className="name"
    >
      Nice Product
    </span>
    <span
      className="price-amount"
    >
      20
    </span>
    <span
      className="price-currency"
    >
      EUR
    </span>
    <span
      className="quantity"
    >
      3
    </span>
    <span
      className="total-amount"
    >
      60
    </span>
    <span
      className="total-currency"
    >
      EUR
    </span>
  </div>
  <div
    className="quote-total"
  >
    <span
      className="amount"
    >
      60
    </span>
    <span
      className="currency"
    >
      EUR
    </span>
  </div>
</div>
`;
